File: brakingRecords/breakingRecords.js
const breakingRecords = (arr) => {
  let totalMin = 0;
  let totalMax = 0;
  let minAux = arr[0];
  let maxAux = arr[0];

  for (let i = 0; i < arr.length; i++) {
    if (arr[i] < minAux) {
      totalMin += 1;
      minAux = arr[i];
    } else if (maxAux < arr[i]) {
      totalMax += 1;
      maxAux = arr[i];
    }
  }

  const totalScores = [totalMax, totalMin];
  return totalScores;
};

const scores = [12, 24, 10, 24];
console.log(breakingRecords(scores));

File: brakingRecords/BreakingRecords.java
package brakingRecords;

import java.util.List;

import java.util.ArrayList;

public class BreakingRecords{
   
    public static List<Integer> breakingRecords (List<Integer> scores){
    int totalMin = 0;
    int totalMax = 0;
    int minAux = scores.get(0);
    int maxAux = scores.get(0);

    for(int score : scores){
        if(minAux>score){totalMin += 1; minAux=score;}
        else if (maxAux<score){totalMax += 1; maxAux=score;}
    }
    List<Integer> TotalScore =  new ArrayList<>();
    TotalScore.add(totalMax);
    TotalScore.add(totalMin);
     
    return TotalScore; 
    }

public static void main(String[] args){
        List<Integer> scores = new ArrayList<>();
        scores.add(12);
        scores.add(24);
        scores.add(10);
        scores.add(24);
        breakingRecords(scores);
    } 
}